lib/bugsnag/notification.rb

```diff
@@ -181,7 +181,7 @@ def deliver
         Bugsnag.warn "No API key configured, couldn't notify"
         return
       elsif api_key !~ API_KEY_REGEX
-        Bugsnag.warn "Your API key (#{api_key}) is not valid, couldn't notify"
```

Comment:
This instance should probably stay; since the API key does not match the required format, its useful to compare the logged output to your actual key to correct typos. There is no data exposure for logging an incorrect key.